hitl/run.sh

```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+
+# Title:
+#   Stabilizer hardware-in-the-loop (HITL) test script.
+#
+# Description:
+#   This shell file is executed by the hardware runner in Quartiq's office to exercise the various
+#   hardware aspects of Stabilizer.
+
+# Enable shell operating mode flags.
+set -eux
+
+# Set up python for testing
+python3 -m venv --system-site-packages py
+source py/bin/activate
```

Comment:
`source` is a bashism. Use `.` or better yet, in this case directly call the desired command:

`py/bin/pip install ...`
and
`py/bin/python miniconf.py ...`

Reply:
Is there a real reason to go through these steps instead of just using `bash`? Github actions always uses bash and this is already specific to our GHA setup given the hostnames in use.

Reply:
I'm more worried about the reproducibility if you source. The `py` env isn't deleted and recreated and at some point there might be a `ping` in `py/bin` because of some mess-up with `requirements.txt`. That will completely confuse everyone. Admittedly a bit unlikely but I regularly see python environments breaking for users exactly like this because they are `sourced` on top of each other or the user forgot to activate/deactivate or had that in `bashrc`/`profile` etc: Then you call `pip` thinking you call it in that environment but in fact you are installing in some other environment.

Reply:
tl;dr I'm more worried about `source` than about the type of shell.

Reply:
I'm also not insisting. Just explaining my past experience and the trouble some users find themselves in with python environments and changing PATHs like this.

Reply:
The environment is automatically cleaned up when GHA completes execution - each `run` step is executed in a temporary shell script, so any environment changes exist purely in the GHA run step and are automatically cleaned up on exit.